src/core/qgsactionmanager.cpp

```diff
@@ -56,21 +57,68 @@ QUuid QgsActionManager::addAction( QgsAction::ActionType type, const QString &na
 
 void QgsActionManager::addAction( const QgsAction &action )
 {
+  QgsDebugMsg( "add action " + action.name() );
   mActions.append( action );
+  if ( mLayer && mLayer->dataProvider() && !action.notificationMessage().isEmpty() )
+  {
+    mLayer->dataProvider()->setListening( true );
+    if ( !mOnNotifyConnected )
+    {
+      QgsDebugMsg( "connecting to notify" );
+      connect( mLayer->dataProvider(), &QgsDataProvider::notify, this, &QgsActionManager::onNotifyRunActions );
+      mOnNotifyConnected = true;
+    }
+  }
+}
+
+void QgsActionManager::onNotifyRunActions( const QString &message )
+{
+  for ( const QgsAction &act : qgsAsConst( mActions ) )
+  {
+    if ( !act.notificationMessage().isEmpty() && act.notificationMessage() == message )
+    {
+      if ( !act.isValid() || !act.runable() )
+        continue;
+
+      QgsExpressionContext context = createExpressionContext();
+
+      Q_ASSERT( mLayer ); // if there is no layer, then where is the notification coming from ?
+      context << QgsExpressionContextUtils::layerScope( mLayer );
```

Comment:
Wouldn't it be very handy to also add make the notification command available in the expression context?

Reply:
clearly, this would allow to use the message in the function, I was looking for a way to to that, thanks.

How do you do that ? 

I was considering regex matching instead of '==' for filtering, but @mhugo mentioned matching the beginning of the message only (like a header), that would allow to send complex messages (like a list of modified features) and, with the message in the context, this would be really powerful.

Reply:
jumping in... I have been secretly hoping for regex support here, but was afraid to get banned ;)

Reply:
@3nids +1 for regex

Reply:
@luipir, @3nids sorry but which kind of regex and which regex engine are we talking about ? ;)

With the last commit, the notification message is filtered on the beginning of the message, so it's nice because the user just has to specified `my_header` for the action, and anything beginning with `my_header` will trigger the action. I think I prefer that to  `my_header.*` or `my_header%`.

Note, you have a limitation of the message length (8000 bytes by default) so nothing too fancy in there.

Reply:
I suppose @3nids was talking to add also a regexp for topic action activation => not only need to match exactly but match some regexp => from the db side can be se sent namespaced notifications.
About size limitation it is referred to paylod (e.g. the value sent with the event) not related with the topic name.

Reply:
Not sure I understand that clearly; 
@luipir what is what you call  "topic action activation" ? Is that the Channel name of the notify you are talking about? Currently it's fixed to qgis, and we thought that the payload offered lot's of flexibility. 
Current header style can be matched and then pass any free text up to 8000 bytes. Any JSON, XML or whatever free text can be send and then filtered in the action code. 
Maybe some basic examples could help.

Reply:
sorry I used the ROS terminology... topic = channel name